nuxeo-features/nuxeo-platform-rendition/nuxeo-platform-rendition-core/src/main/java/org/nuxeo/ecm/platform/rendition/lazy/AbstractRenditionBuilderWork.java

```diff
@@ -110,24 +110,31 @@ public void work() {
 
         log.debug("Starting rendition computation.");
         List<Blob> blobs = doComputeRendition(session, doc, def);
-        updateAndCompleteStoreEntry(blobs);
+        updateAndCompleteStoreEntry(doc, blobs);
     }
 
     @Override
     public void cleanUp(boolean ok, Exception e) {
-        super.cleanUp(ok, e);
         if (ok) {
+            super.cleanUp(ok, e);
             return;
         }
+
+        // Fetch document and load its data before cleaning up which closes the session
+        DocumentModel doc = session.getDocument(docRef);
+        doc.detach(true);
```

Comment:
Instead of detaching the full doc (which is costly), you can just compute the `sourceDocumentModificationDate` at this point, which is all you need from the document.